File: project/dahee-onboarding/survey-core/src/main/java/com/survey/core/enums/QuestionType.java
package com.survey.core.enums;

/***
 * 미리 허용된 타입만 쓰도록 강제, 잘못된 타입 입력시 컴파일/파싱 단계 오류
 */
public enum QuestionType {
    SHORT_TEXT, //단답형
    LONG_TEXT, //장문형
    SINGLE_CHOICE, //단일 선택
    MULTI_CHOICE //다중 선택
}

File: project/jihochoi-onboarding/QueryBox/src/main/kotlin/com/wlghsp/querybox/domain/survey/QuestionType.kt
package com.wlghsp.querybox.domain.survey

enum class QuestionType {
    SHORT_TEXT, LONG_TEXT, SINGLE_CHOICE, MULTIPLE_CHOICE;

    // 단일 or 다중 선택 리스트 인지 확인
    fun isChoice(): Boolean = this == SINGLE_CHOICE || this == MULTIPLE_CHOICE
}
